py/plugins/gcc.py

```
@@ -139,11 +150,32 @@ def handle_args(self, parser, args, props):
             self.enable_sanitize(props, ["libasan"], ["-fsanitize=address"])
 
             # leak checker is currently too picky even for standard libs
-            props.env["ASAN_OPTIONS"] = "detect_leaks=0"
+            props.env["ASAN_OPTIONS"] += ",detect_leaks=0"
+
+            # detect usage of already destroyed stack variables
+            props.env["ASAN_OPTIONS"] += ",detect_stack_use_after_return=1"
 
             # -fsanitize=address does not seem to be supported with -static
             self.flags.remove_flags(["-static"])
 
+            # preload ASAN for %check
+            def run_asan_hook(results, mock, props):
+                cmd = "echo /usr/lib64/libasan.so.*.* > /etc/ld.so.preload"
+                rv = mock.exec_chroot_cmd(cmd)
+                if 0 != rv:
+                    results.error(f"ASAN plug-in: ASAN preloading failed with exit code {rv}")
+                    return rv
+
+                # FIXME: hack
+                extra_env = {"ASAN_OPTIONS": props.env["ASAN_OPTIONS"] + ",verify_asan_link_order=0"}
+                ec = mock.exec_rpmbuild_bi(props, extra_env=extra_env)
+                if 0 != ec:
+                    results.error(f"ASAN plug-in: %install or %check failed with exit code {rv}")
+                return ec
+
+            props.run_check = False
```

Although the csmock plug-in API is undocumented, the plug-ins are expected to either set these flags, or leave them unchanged.  Explicitly clearing the `props.run_check` flag may silently break other plug-ins enabled at the same time.

---

Ok, I'll rework it. However, leaving it enabled won't work since the code was built with ASAN. One solution would be to preload ASAN for everything every time, but that is far from ideal.

---

If a combination of plug-ins or their options does not work together, I think it is better to fail, rather than silently paralyzing other plug-ins.  We already have a few `assert` commands in the plug-ins because of this.

---

Addressed in eda1cf97596e67ecd0f9285dd42834cc0491dc53.